source/DataAccess/DADGenerator.ts

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+import IntegrationConfigFactory from '../IntegrationConfigFactory';
+import { IIntegrationConfig, IntegrationType } from '../IIntegrationConfig';
+import { mkdir, writeFile } from 'fs';
+
+export default class DataAccessDoc {
+
+    private _integrationType;
+    private _queries;
+    private _header: string[] = [];
+    private _footer: string[] = [];
+
+    constructor(integrationType: string) {
```

Comment:
You can lock down the possible values of integration type by using the 'IntegrationType' as the type instead of string. (See IIntegrationConfig.ts)